### PR TITLE
Fixes #6129 (declaration of coercions made compatible with local definitions)

### LINKED ISSUE #6129: Record field coercions broken with := in the arguments
#### Version

8.7.0

#### Description of the problem

This code should work.  It does not:
```coq
Record foo (x:=1) := { foo_nat :> nat }. (* Error: foo_nat is not a function. *)
Record foo2 (x : Type) (y:=1)  := { foo_nat2 :> nat }. (* Error: Cannot recognize foo as a source class of foo_nat. *)
```

### PR DESCRIPTION
**Pull request by herbelin**
Declaration of coercions had actually never been made compatible with local definitions in the premisses of the coercion.

### CODE REVIEW

File: test-suite/bugs/closed/6129.v [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(* Make definition of coercions compatible with local definitions. *)
+
+Record foo (x : Type) (y:=1) := { foo_nat :> nat }.
+Record foo2 (x : Type) (y:=1) (z t: Type) := { foo_nat2 :> nat }.
```
Comment (JasonGross) [inline test-suite/bugs/closed/6129.v]: You're missing a case: this tests when the let-binder is the last argument, and when it's a middle argument, but it doesn't test when the let-binder is the first argument.

**Review by JasonGross — DISMISSED** [review]

File: test-suite/bugs/closed/6129.v (hunk 1 above)
Comment (herbelin) [reply test-suite/bugs/closed/6129.v]: I added a test (it works), but the test is anyway an approximation. We could have tested dependent contexts, presence of section variables, and many various situations. You're right in wanting more tests. Feel free to add as many tests as you want.

Comment (JasonGross) [reply test-suite/bugs/closed/6129.v]: Thanks.  I thought the first/last tests were particularly salient as corner cases because I got different error messages for each of them.

Comment (herbelin) [reply test-suite/bugs/closed/6129.v]: OK, I see.

**Review by gares — CHANGES_REQUESTED** [review]
The code looks good, but I'd like to have the test not only check that the coercions are declared, but also inserted. Could you please add a few invocation to `Check (fun x : foo{1,2,3} => x : nat)`?

**Comment by herbelin** [comment]
@gares: done. (Please feel free to push directly to this branch if you want to add more test.)

**Review by gares — APPROVED** [review]
